File: Backend-node/src/routes/textRoutes.js
const express = require('express');
const router = express.Router();
const multer = require('multer');
const textController = require('../controllers/textController');

// Usa multer para manejar la carga de imágenes
const upload = multer({ storage: multer.memoryStorage() });

router.post('/extract-text', upload.single('image'), textController.extractTextFromImage);
router.post('/translate-text', textController.detectLanguageAndTranslate);
router.post('/get-labels', upload.single('image'), textController.getLabelsFromImage);


module.exports = router;

File: Backend-python/src/routes/textRoutes.py
from flask import Blueprint, request, jsonify
import controllers.textController as textController

textController_bp = Blueprint('textController_bp', __name__, url_prefix = '/api')

@textController_bp.route('/extract-text', methods=['POST'])
def extractTextFromImage():
    return textController.extractTextFromImage()

@textController_bp.route('/translate-text', methods=['POST'])
def detectLanguageAndTranslate():
    return textController.detectLanguageAndTranslate()

@textController_bp.route('/get-labels', methods=['POST'])
def getLabelsFromImage():
    return textController.getLabelsFromImage()